As of sanic 21.3.0 pytest-sanic fails with "TypeError: serve() got an unexpected keyword argument 'main_start'"

Running a stripped down version of the sanic_client example fails when used with current sanic-21.3.2:

```_______________________________________________ ERROR at setup of test_fixture_test_client_get _______________________________________________

loop = <uvloop.Loop running=False closed=False debug=False>, app = Sanic(name="test_sanic_app")
sanic_client = <function sanic_client.<locals>.create_client at 0x7fccc46691f0>

    @pytest.fixture
    def test_cli(loop, app, sanic_client):
>       return loop.run_until_complete(sanic_client(app, protocol=WebSocketProtocol))

tests/test_pytest.py:20:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
uvloop/loop.pyx:1494: in uvloop.loop.Loop.run_until_complete
    ???
.venv/lib/python3.8/site-packages/pytest_sanic/plugin.py:203: in create_client
    await client.start_server()
.venv/lib/python3.8/site-packages/pytest_sanic/utils.py:224: in start_server
    await self._server.start_server()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pytest_sanic.utils.TestServer object at 0x7fccc465e850>

    async def start_server(self):
        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.socket.bind((self.host, 0))
        self.port = self.socket.getsockname()[1]

        # server settings
        server_settings = self.app._helper(
            host=self.host, port=self.port,
            ssl=self.ssl, sock=self.socket,
            loop=self.loop, protocol=self.protocol,
            backlog=self.backlog, run_async=True)

        # clean up host/port.
        # host/port should not be used.
        server_settings['host'] = None
        server_settings['port'] = None

        # Let's get listeners
        self.before_server_start = server_settings.get('before_start', [])
        self.after_server_start = server_settings.get('after_start', [])
        self.before_server_stop = server_settings.get('before_stop', [])
        self.after_server_stop = server_settings.get('after_stop', [])

        # Trigger before_start events
        await trigger_events(self.before_server_start, self.loop)

        # Connections
        server_settings["connections"] = self.connections

        # start server
>       self.server = await serve(**server_settings)
E       TypeError: serve() got an unexpected keyword argument 'main_start'
```

Adding something like:

```python
        main_start = server_settings.pop("main_start", None)
        main_stop = server_settings.pop("main_stop", None)
        if main_start or main_stop:
            warnings.warn(
                "Listener events for the main process are not available "
                "with start_server()"
            )
```
in utils.py should fix it.

Support for Sanic-21 additionally requires dropping python-3.6 as it is on >=3.7 - allowing uvloop-0.15.2 instead of 0.14.0.

I can create a PR if you want me to...

--- a/pytest_sanic/utils.py
+++ b/pytest_sanic/utils.py
@@ -97,6 +97,9 @@ class TestServer:
         self.before_server_stop = server_settings.get('before_stop', [])
         self.after_server_stop = server_settings.get('after_stop', [])
 
+        server_settings.pop("main_start", None)
+        server_settings.pop("main_stop", None)
+
         # Trigger before_start events
         await trigger_events(self.before_server_start, self.loop)
 

